clientv3/sync/syncer.go

```diff
@@ -133,16 +133,9 @@ func (s *syncer) SyncUpdates(ctx context.Context) clientv3.WatchChan {
 }
 
 func incr(bs []byte) []byte {
-	c := int8(1)
-	for i := range bs {
-		j := len(bs) - i - 1
-		n := int8(bs[j])
-		n += c
-		bs[j] = byte(n)
-		if n == 0 {
-			c = 1
-		} else {
-			c = 0
+	for i := len(bs) - 1; i >= 0; i-- {
+		bs[i] = bs[i] + 1
```

Comment:
This can end up with trailing 0's; I don't think it's correct. There's a simpler version in contrib/recipes/range.go (prefixEnd).

I'll post a PR to make the prefix stuff part of the API.